src/LevelZeroDevicePool.hpp

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (c) 2015, 2016, 2017, 2018, 2019, 2020, Intel Corporation
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ *     * Redistributions of source code must retain the above copyright
+ *       notice, this list of conditions and the following disclaimer.
+ *
+ *     * Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in
+ *       the documentation and/or other materials provided with the
+ *       distribution.
+ *
+ *     * Neither the name of Intel Corporation nor the names of its
+ *       contributors may be used to endorse or promote products derived
+ *       from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY LOG OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef LEVELZERODEVICEPOOL_HPP_INCLUDE
+#define LEVELZERODEVICEPOOL_HPP_INCLUDE
+
+#include <vector>
+#include <string>
+#include <cstdint>
+
+#include "geopm_sched.h"
+
+namespace geopm
+{
+
+    class LevelZeroDevicePool
+    {
+        public:
+            LevelZeroDevicePool() = default;
+            virtual ~LevelZeroDevicePool() = default;
+            /// @brief Number of accelerators on the platform.
+            /// @return Number of LevelZero accelerators.
+            virtual int num_accelerator(void) const = 0;
+            /// @brief Get the LevelZero device core clock rate
+            ///        in MHz.
+            /// @param [in] accel_idx The index indicating a particular
+            ///        accelerator.
+            /// @return Accelerator device core clock rate in MHz.
+            virtual double frequency_status_gpu(unsigned int accel_idx) const = 0;
+            virtual double frequency_status_mem(unsigned int accel_idx) const = 0;
```

Comment:
Header descriptions required.  A large number of these functions will be moved into another helper class as part of the planned refactor, so bundling those changes together.